### ISSUE TITLE
Types of votes and Proposal States 

### ISSUE THREAD

**Issue by HabacucMX** [zenbitETH/zengo#24]
### Types of moderator votes
There are two types of moderator votes:
1. ProposalVote (1 vote per moderator to update the states of the proposals, updates require 60% of moderator votes to pass)
2. Funding Vote (Plural Voting Points to distribute the funds quadratically)
 
 ### Proposal States:
 The proposals have **five main states:**

> **1. Verificación (Verification) [ProposalVote]**
> _**2. Fondeo (Funding) [FundingVote]**_  
> **3. Seguimiento (Following) [ProposalVote]**  
> **4. Finalizada (Completed) [ProposalVote]**  
> **5. Rechazada (Rejected) [ProposalVote]** 

 
-  The first stage (Verification) comprises four substages and will have a dedicated screen on the proposal details page:

> 1. Verificación (Verification) [ProposalVote]
>      a. Nueva propuesta (New proposal) 
>      b. Evidencia Municipal (Municipal evidence) 
>      c. Evidencia Estatal (State evidence) 
>      d. Evidencia PNT (Federal evidence)

If the address is moderator, ProposalVote buttons should appear in **panel 3.** Those buttons should show the current votes and trigger the state update automatically if 60% of moderators vote on a single option. 
![Screenshot 2023-08-19 at 22 55 51](https://github.com/zenbitETH/zengo/assets/57845110/105b2c70-65ac-4233-8699-8ebef6b60559)

- The second (funding) will be displayed on the fondeoPlural page:
![Screenshot 2023-08-19 at 23 33 07](https://github.com/zenbitETH/zengo/assets/57845110/03fe0190-8a4a-4b28-a2ac-09f9383a9b44)

- The third (following) will be displayed on the segumiento page:
![Screenshot 2023-08-19 at 23 33 48](https://github.com/zenbitETH/zengo/assets/57845110/79ee8eb0-8dbd-4f53-b3e1-5f3a9c23bf3f)

- If a proposal is completed or rejected, the state will be shown on the proposal state in panel three from proposal detail, proposal dashboard, and following.
 
![Screenshot 2023-08-19 at 23 36 13](https://github.com/zenbitETH/zengo/assets/57845110/1283c9bd-3ed3-4041-9b3e-c7c231f82227)
![Screenshot 2023-08-19 at 23 38 17](https://github.com/zenbitETH/zengo/assets/57845110/6e9fcca2-7693-4c16-97f6-6a2e8c053ae3)

**Comment by Markkos89** [comment]
@shyam-patel-kira how do you think the states can be "stored" in the proposal? it's a property in the struct, right?}

with my low solidity knowledge I can imagine something like this, let's define it together:

This is based on what @HabacucMX is explaining (leaving the UI aside)

```solidity
struct Proposal {
        uint256 id;
        string title;
        string description;
        address submitter;
        bool isVerified; // this is not used I think (??
        uint256 votes;
        proposalState: string
    }
```

and the states (for the sake of throw the first stone and start building this up :rofl: ):

```solidity
enum proposalStates {
        Verification,
        Funding,
        Following,
        Completed,
        Rejected
    }
```

After writing this, I can see that the states are mixed with the page the proposal should be displayed, perhaps? correct me if I'm wrong please...

**Comment by Markkos89** [comment]
After reviewing this flow with @HabacucMX we were able to detect and define the states and stages of the proposals, and we resolved to use a code to represent it.

So, the flow/route each proposal will go trought is at follows:

1. Verification
1a. New proposal
1b. Municipal evidence
1c. State evidence
1d. Federal evidence
2. Funding
3. Following
4. Completed

And the only steps in where the proposals can be Rejected are the step 1, 1a, 1b, 1c or in 1d.
If any proposal arrives until the second step (of Funding) it will complete the flow/route until be Completed.

- In the smart contract, **the Proposal struct needs to have a `state` propoerty** which will be an enum (to be defined by Shyam) that can have this values:

1, 1a, 1ar, 1b, 1br, 1c, 1cr, 1d, 1dr, 2, 3, or 4.

Where the extra `r` letter represents that the proposal was `Rejected` at this step. This will help us to actually know until which step each proposal went trought, and to not have to store all the words.
The frontend will decode this representation code and use it.

- **The Proposal struct also needs to have an boolan property** representing:

true = it's an ongoing proposal. open
false = it's closed, it was rejected or it finished

@shyam-patel-kira the bolded text is actually what is need to be done, the rest is the explanation of it :smiley: Let us know your thoughts!
